src/main/resources/static/vulnerableApp.js

```diff
@@ -70,75 +70,86 @@ function _callbackForInnerMasterOnClickEvent(
   };
 }
 
-function handleFirstElementAutoSelection(vulnerableAppEndPointData) {
-  if (vulnerableAppEndPointData.length > 0) {
-    vulnerabilitySelected = vulnerableAppEndPointData[0]["Name"];
-    detailTitle.innerHTML = vulnerableAppEndPointData[0]["Description"];
-    let isFirst = true;
-    for (let key in vulnerableAppEndPointData[0]["Detailed Information"]) {
-      let column = document.createElement("div");
-      column.id = "0." + key;
-      let textNode = document.createTextNode(
-        vulnerableAppEndPointData[0]["Detailed Information"][key]["Level"]
-      );
-      column.appendChild(textNode);
-      column.className = "inner-master-item";
-      column.addEventListener(
-        "click",
-        _callbackForInnerMasterOnClickEvent(
-          vulnerableAppEndPointData,
-          0,
-          key,
-          vulnerabilitySelected
-        )
-      );
-      if (isFirst) {
-        column.click();
-        isFirst = false;
-      }
-      innerMaster.appendChild(column);
+function getVariant(detailedInformation) {
+  return detailedInformation["Variant"] === "SECURE";
+}
+
+function getSvgVariant(isSecure) {
+  let svg = document.createElement("img");
+  svg.classList.add("vector");
+  svg.setAttribute("src", "vectors/" + (isSecure ? "secure" : "unsecure") + ".svg");
+
+  return svg;
+}
+
+function createColumn(detailedInformationArray, key) {
+  let detailedInformation = detailedInformationArray[key];
+  let isSecure = getVariant(detailedInformation);
+
+  let column = document.createElement("div");
+  column.id = "0." + key;
+  column.appendChild(getSvgVariant(isSecure));
+  column.appendChild(document.createTextNode(detailedInformation["Level"]));
+  column.classList.add("inner-master-item");
+
+  if (isSecure) {
+    column.classList.add("secure-vulnerability");
+  }
+
+  return column;
+}
+
+function appendNewColumn(vulnerableAppEndPointData, id) {
+  let detailedInformationArray = vulnerableAppEndPointData[id]["Detailed Information"];
+  let isFirst = true;
+
+  for (let key in detailedInformationArray) {
+    if (!detailedInformationArray.hasOwnProperty(key)) {
+      continue;
     }
+    let column = createColumn(detailedInformationArray, key);
+    column.addEventListener(
+      "click",
+      _callbackForInnerMasterOnClickEvent(
+        vulnerableAppEndPointData,
+        id,
+        key,
+        vulnerabilitySelected
+      )
+    );
+    if (isFirst) {
+      column.click();
+      isFirst = false;
+    }
+    innerMaster.appendChild(column);
   }
 }
 
+function handleElementAutoSelection(vulnerableAppEndPointData, id = 0) {
+  if (!vulnerableAppEndPointData.length) {
+    return;
+  }
+
+  if (id) {
```

Comment:
as id is an integer so please add a proper condition like `id === 0`

Reply:
Resolved in 422b8d3